build_tools/embed_data/generate_embed_data_main.cc

```diff
@@ -140,10 +140,11 @@ static bool GenerateImpl(const std::string& identifier,
 #define IREE_DATA_ALIGNAS_PTR __declspec(align(8))
 #else
 #include <stdalign.h>
-#define IREE_DATA_ALIGNAS_PTR alignas(alignof(void*))
+// Default set to 64-bit alignment.
+#define IREE_DATA_ALIGNAS_PTR alignas(8)
```

Comment:
I think we'll want this as a higher value (not just pointer width, so this whole ifdef stuff can go away) for other devices that have alignment requirements - 64 bytes is what we internally pad the vmfb contents to, so matching that in the binary hosting the vmfb would be nice.

Reply:
I can set the default to 64B (512b), but for 32-bit system it may be too much padding, and we probably want to define `IREE_DATA_ALIGNAS_PTR` for that architecture explicitly. 

I also tried to call `IREE_HOST_SIZE_T` here but then `generate_embed_data` would be depending on `runtime/src/iree/base/config.h`, which would be an ugly solution.

Reply:
Yeah, something that we could add a flag to override in the compiler but for now saving a few bytes anywhere else would be easier :P We can hardcode it to 64 for now and when we come around to adding the compiler flag we can figure out how to plumb that through. The trick is having the compiler flag for a particular artifact somehow passed to the embed_data tool and that makes my head hurt - I'd sooner say 50 bytes don't matter at the point you're using a flatbuffer instead of emitc :)

Reply:
Fair point. However, I tried to define `IREE_DATA_ALIGNAS_PTR` explicitly and then some other part of the build breaks. I will fix those with another PR.